cupy/core/_routines_math.pxd

```diff
@@ -10,10 +10,13 @@ cdef ndarray _ndarray_prod(ndarray self, axis, dtype, out, keepdims)
 cdef ndarray _ndarray_sum(ndarray self, axis, dtype, out, keepdims)
 cdef ndarray _ndarray_cumsum(ndarray self, axis, dtype, out)
 cdef ndarray _ndarray_cumprod(ndarray self, axis, dtype, out)
+cdef ndarray _ndarray_nansum(ndarray self, axis, dtype, out, keepdims)
+cdef ndarray _ndarray_nanprod(ndarray self, axis, dtype, out, keepdims)
 cdef ndarray _ndarray_clip(ndarray self, a_min, a_max, out)
 
 cdef ndarray scan(ndarray a, ndarray out=*)
 cdef object _sum_auto_dtype
+cdef object _nansum_auto_dtype
```

Comment:
This line seems not needed.

Reply:
This is used in `_routines_math.pyx`, just like `_sum_auto_dtype`, such as in https://github.com/cupy/cupy/pull/2252/files#diff-aa843213093afe1f96359cbe1628c4cbR89.

Reply:
Cython uses `.pxd` files like as C header files, so we have to write a declaration if the function is called from outside (e.g, `_sum_auto_dtype` is called from [here](https://github.com/cupy/cupy/blob/master/cupy/core/_routines_indexing.pyx#L879)), but `_nansum_auto_dtype` is not called only from inside.

Reply:
Thanks for the clarification. Indeed I'm not yet very familiarized with the internals of CuPy. This is done now.